packages/tracing/src/browser/browsertracing.ts

```diff
@@ -126,8 +194,14 @@ export class BrowserTracing implements Integration {
     }
 
     const hub = this._getCurrentHub();
-    logger.log(`[Tracing] starting ${ctx.op} idleTransaction on scope with context:`, ctx);
-    return startIdleTransaction(hub, ctx, idleTimeout, true) as TransactionType;
+    logger.log(`[Tracing] starting ${ctx.op} idleTransaction on scope`);
+    const idleTransaction = startIdleTransaction(hub, ctx, idleTimeout, true);
+    idleTransaction.registerBeforeFinishCallback(adjustTransactionDuration(secToMs(maxTransactionDuration)));
+    idleTransaction.registerBeforeFinishCallback(transaction => {
+      this._metrics.addPerformanceEntires(transaction);
+    });
```

Comment:
Maybe to unify it with L199 syntax? (not sure about `this` context here, would have to verify it. Or we can make L199 accept arrow function instead to make it more "obvious" instead of curried function, your call
```suggestion
    idleTransaction.registerBeforeFinishCallback(this._metrics.addPerformanceEntires);
```

Reply:
Yeah this issue here was with the `this` context. 

I refactored to something like this:

```ts
idleTransaction.registerBeforeFinishCallback((transaction, endTimestamp) => {
  this._metrics.addPerformanceEntires(transaction);
  adjustTransactionDuration(secToMs(maxTransactionDuration), transaction, endTimestamp);
});
```